### ISSUE TITLE
Not work well with pnpm

### ISSUE THREAD

**Issue by larryro** [payloadcms/payload#3535]
### Link to reproduction

https://payloadcms.com/docs/getting-started/installation#adding-to-an-existing-app

### To Reproduce

Install the payload using pnpm and use webpack as the bundler.

### When serve the app, we can see the error below:
```
ERROR in main
Module not found: Error: Can't resolve '/workspaces/tale/node_modules/.pnpm/@payloadcms+bundler-webpack@1.0.2_@swc+core@1.3.90_esbuild@0.18.20/node_modules/payload/dist/admin' in '/workspaces/tale'
```    
  

### Then go to the corresponding directory, and you will only find the "payloadcms" directory. There is no "payload" directory.   

![image](https://github.com/payloadcms/payload/assets/5787117/83b885d2-79c7-4ae3-b91d-78e73599c647)
 

### If specify the webpack entry point in the code manual, everything works well   

![image](https://github.com/payloadcms/payload/assets/5787117/d410858e-acf7-41d0-9575-3d181c94f560)

### Payload Version

2.0.3

### Adapters and Plugins

 db-mongodb, bundler-webpack

**Comment by jmikrut** [comment]
Probably need to add `@swc/core` to the `peerDependencies` of `bundler-webpack`. We're on it!

**Comment by denolfe** [comment]
Fixed by 65cdf6bfd. Had to update aliasing to be more intelligent about finding the node_modules dir.
